pkg/workflows/builder.go

```diff
@@ -0,0 +1,73 @@
+package workflows
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// 1. Capability defines JSON schema for inputs and outputs of a capability.
+// Trigger: triggerOutputType := workflowBuilder.addTrigger(DataStreamsTrigger.Config{})
+// Adds metadata to the builder. Returns output type.
+// 2. Consensus: consensusOutputType := workflowBuilder.addConsensus(ConsensusConfig{
+// 	Inputs: triggerOutputType,
+// })
+
+type Workflow struct {
+	spec *workflowSpecYaml
```

Comment:
I would build the underlying WorkflowSpec here, and make that into a proto.  You can move the validation from workflowSpecYaml to WorkflowSpec as well.  This would make it more flexible.